internal/experimental/registry/client.go

```diff
@@ -539,3 +572,53 @@ func PushOptStrictMode(strictMode bool) PushOption {
 		operation.strictMode = strictMode
 	}
 }
+
+// Tags provides a sorted list all semver compliant tags for a given repository
+func (c *Client) Tags(ref string) ([]string, error) {
+	parsedReference, err := registry.ParseReference(ref)
+	if err != nil {
+		return nil, err
+	}
+
+	repository := registryremote.Repository{
+		Reference: parsedReference,
+		Client:    c.registryAuthorizer,
+	}
+
+	var registryTags []string
+
+	for {
+		registryTags, err = registry.Tags(ctx(c.out, c.debug), &repository)
+		if err != nil {
+			// Fallback to http based request
+			if !repository.PlainHTTP && strings.Contains(err.Error(), "server gave HTTP response") {
+				repository.PlainHTTP = true
+				continue
+			}
+			return nil, err
+		}
+
+		break
+
+	}
+
+	var tagVersions []*semver.Version
+	for _, tag := range registryTags {
+		tagVersion, err := semver.StrictNewVersion(tag)
```

Comment:
```
tagVersion, err := semver.StrictNewVersion(strings.ReplaceAll(tag, "_", "+"))
```

Need to make sure the _ has been turned back into a + before parsing the version

Reply:
Ah, thanks. I even left a note under "special notes" https://github.com/helm/helm/pull/10521 so we didn’t forget to make sure to add back to this PR. Guess what, I forgot 😛